homeassistant/components/vizio/__init__.py

```diff
@@ -37,11 +44,14 @@ def validate_auth(config: ConfigType) -> ConfigType:
     vol.Optional(CONF_ACCESS_TOKEN): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS): vol.All(
-        cv.string, vol.Lower, vol.In(["tv", "soundbar"])
+        cv.string, vol.Lower, vol.In([DEVICE_CLASS_TV, DEVICE_CLASS_SPEAKER])
```

Comment:
Why do we need to change this user interface? We could just use the mapping dict we created and lookup the home assistant device class from vizio device class.

Reply:
The `pyvizio` module was updated to use `speaker` as the device class instead of `soundbar` as well since it is more generic. The purpose of the dict is to ensure that any changes to the values of `DEVICE_CLASS_TV` and `DEVICE_CLASS_SPEAKER` in HA don't break the integration. If you think this is overkill, I can remove this dict and use the HA device class directly.

Reply:
Ok, if it makes sense long term it's ok.

We should keep the mapping dict. We should never couple two different naturally independent APIs so that they become dependent. If we would make them dependent we risk breakage if one changes.

Reply:
👍 in that case I believe I have resolved all of the issues you identified